### PR TITLE
[da-vinci][server] Fixed a small perf regression when handling compute request

### PR DESCRIPTION
**Pull request by gaojieliu**
In a previous schema optimization, Venice Server/DaVinci could pass an inconsistent schema id from the reused record when decoding the binary payload from the database.
When mismatch happens, object reuse won't work, and Avro deserializer will create a new object for every call.
For Server, this could happen in the following setup:
1. The user evolves the value schema, which would also trigger a new superset schema.
2. The new value/superset schema won't be used by any data entries.
3. Client will continue to use the locally-known latest value schema.
4. The mismatch will happen in Venice Server when handling such compute request.

For DaVinci, the chance of mismatch is very low since the request will be executed locally, and that is the reason this PR uses a lightweight change to make schema & schema id consistent when serving compute request.

This code change also fixed a regression introduced by client compute support, and
it will handle the scenario if an error happens before handling any header/data received
from the backend. Currently, this seems only happening in integration test, which is
using `HttpTransportClient`, where the streaming impl is different from `D2TransportClient`.

This code change also enhances a test utility: `ErrorCountAppender` to handle an error log without a valid throwable or a throwable without a valid message.

## How was this PR tested?
CI

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: services/venice-server/src/test/java/com/linkedin/venice/listener/ErrorCountAppender.java [hunk 1]
```diff
@@ -57,7 +57,9 @@ public ErrorCountAppender build() {
 
   @Override
   public void append(LogEvent event) {
-    if (event.getLevel().equals(Level.ERROR) && event.getThrown().getLocalizedMessage().contains(exceptionMessage)) {
+    if (event.getLevel().equals(Level.ERROR) && event.getThrown() != null
```
Comment (FelixGV) [inline services/venice-server/src/test/java/com/linkedin/venice/listener/ErrorCountAppender.java:60]: Could you mention this change also in the commit message?

**Review by FelixGV — COMMENTED** [review]
Thanks Gaojie! The change LGTM... but there is an integration test failure, so we should look at that first...

**Comment by gaojieliu** [comment]
> Thanks Gaojie! The change LGTM... but there is an integration test failure, so we should look at that first...
Yes, I found a regression in client compute, which seems not serious in prod, but troublesome in integration test since integration test is normally using http client instead of d2 and http client's streaming impl is different from d2 client.
Fixed it in the latest code .

File: services/venice-server/src/test/java/com/linkedin/venice/listener/ErrorCountAppender.java (hunk 1 above)
Comment (gaojieliu) [reply services/venice-server/src/test/java/com/linkedin/venice/listener/ErrorCountAppender.java:60]: Updated.

File: build.gradle [hunk 2]
```diff
@@ -415,7 +415,7 @@ subprojects {
     }
 
     retry {
-      maxRetries = 4 // 5 attempts in total
```
Comment (majisourav99) [inline build.gradle:418]: not part of the change?

Comment (gaojieliu) [reply build.gradle:418]: yeah, I missed it, and I will revert it.

**Review by FelixGV — APPROVED** [review]
Thanks!

**Review by majisourav99 — APPROVED** [review]
Thanks Gaojie. LGTM
